--- modules/libfabric/prov/hook/src/hook_domain.c 
+++ modules/libfabric/prov/hook/src/hook_domain.c 
@@ -105,7 +105,7 @@
 {
        /*
         * called from the base provider, ep_fid is the base ep, and
         * it's fid context is the hook ep.
         */
        struct hook_ep *ep = (struct hook_ep *)ep_fid->fid.context;

-        return (*ep->domain->base_credit_handler)(&ep->ep, credits);
+        return (*ep->domain->base_credit_handler)(&ep->ep, (uint64_t)credits);
 }

@@ -117,8 +117,8 @@
 static void hook_set_send_handler(struct fid_domain *domain_fid,
-                ssize_t (*credit_handler)(struct fid_ep *ep, size_t credits))
+                ssize_t (*credit_handler)(struct fid_ep *ep, uint64_t credits))
 {
        struct hook_domain *domain = container_of(domain_fid,
                                                  struct hook_domain, domain);

-        domain->base_credit_handler = credit_handler;
-        domain->base_ops_flow_ctrl->set_send_handler(domain->hdomain,
+        domain->base_credit_handler = credit_handler;
+        domain->base_ops_flow_ctrl->set_send_handler(domain->hdomain,
                                                     hook_credit_handler);
 }
