--- modules/libfabric/prov/hook/src/hook_domain.c
+++ modules/libfabric/prov/hook/src/hook_domain.c
@@ -102,7 +102,7 @@
        .regattr = hook_mr_regattr,
 };

-static ssize_t hook_credit_handler(struct fid_ep *ep_fid, size_t credits)
+static ssize_t hook_credit_handler(struct fid_ep *ep_fid, uint64_t credits)
 {
        /*
         * called from the base provider, ep_fid is the base ep, and
@@ -113,7 +113,7 @@
        struct hook_ep *ep = (struct hook_ep *)ep_fid->fid.context;

-        return (*ep->domain->base_credit_handler)(&ep->ep, credits);
+        return (*ep->domain->base_credit_handler)(&ep->ep, (uint64_t)credits);
 }

@@ -117,7 +117,7 @@
 static void hook_set_send_handler(struct fid_domain *domain_fid,
-                ssize_t (*credit_handler)(struct fid_ep *ep, size_t credits))
+                ssize_t (*credit_handler)(struct fid_ep *ep, uint64_t credits))
 {
        struct hook_domain *domain = container_of(domain_fid,
                                                 struct hook_domain, domain);

